### PR TITLE
Move services out of :persistent-cache

### PR DESCRIPTION
**Pull request by lptr**
The `:persistent-cache` subproject is about the API and implementation for the low-level caches. Scopes and higher level caches don't belong here, and should be moved to where they are used.

Requirement for https://github.com/gradle/gradle/pull/28582.

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: platforms/core-execution/build-cache-local/build.gradle.kts [hunk 1]
```diff
@@ -6,13 +6,10 @@ plugins {
...
 
 dependencies {
-    api(project(":build-cache"))
-    api(libs.inject)
-
     api(project(":base-annotations"))
     api(project(":base-services"))
+    api(project(":build-cache"))
     api(project(":build-cache-spi"))
-    api(project(":core-api"))
```
Comment (lptr) [inline platforms/core-execution/build-cache-local/build.gradle.kts:15]: 💭 This is important: our local build cache implementation doesn't depend on the whole public Gradle API anymore.

File: platforms/core-execution/persistent-cache/build.gradle.kts [hunk 2]
```diff
@@ -28,11 +28,10 @@ dependencies {
     api(project(":messaging"))
     api(project(":native"))
     api(project(":files"))
-    api(project(":logging"))
```
Comment (lptr) [inline platforms/core-execution/persistent-cache/build.gradle.kts:31]: 💭 This is also important: `:persistent-cache` has no business with logging, and now that's actually the case.

File: platforms/core-execution/persistent-cache/src/test/groovy/org/gradle/cache/internal/DefaultCacheBuilderTest.groovy [hunk 3]
```diff
@@ -26,19 +26,19 @@ import static org.gradle.cache.FileLockManager.LockMode.OnDemand
 import static org.gradle.cache.FileLockManager.LockMode.Shared
 import static org.gradle.cache.internal.filelock.DefaultLockOptions.mode
 
-class DefaultUnscopedCacheBuilderFactoryTest extends Specification {
+class DefaultCacheBuilderTest extends Specification {
```
Comment (lptr) [inline platforms/core-execution/persistent-cache/src/test/groovy/org/gradle/cache/internal/DefaultCacheBuilderTest.groovy:29]: 💭 `UnscopedCacheBuilderFactory` is just a thin service wrapper of the only meaningful implementation of `CacheBuilder`. Now we test the meaningful thing.

File: platforms/core-execution/snapshots/build.gradle.kts [hunk 4]
```diff
@@ -14,14 +14,14 @@ errorprone {
...
     api(project(":files"))
     api(project(":functional"))
     api(project(":hashing"))
 
-    implementation(project(":base-annotations"))
+    api(libs.guava)
+    api(libs.jsr305)
 
-    implementation(libs.guava)
-    implementation(libs.jsr305)
```
Comment (lptr) [inline platforms/core-execution/snapshots/build.gradle.kts:24]: 💭 All of the changes to Guava, JSR-305 and `:base-annotations` in this PR is just noise.

File: subprojects/core/src/main/java/org/gradle/cache/internal/DefaultUnscopedCacheBuilderFactory.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+package org.gradle.cache.internal;
+
+import org.gradle.cache.CacheBuilder;
+import org.gradle.cache.UnscopedCacheBuilderFactory;
+
+import java.io.File;
+
+public class DefaultUnscopedCacheBuilderFactory implements UnscopedCacheBuilderFactory {
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/cache/internal/DefaultUnscopedCacheBuilderFactory.java:23]: 💭 I don't know why Git didn't recognize this as a move...

File: gradle.properties [hunk 6]
```diff
@@ -26,7 +26,6 @@ systemProp.dependency.analysis.test.analysis=false
 
 # List of project that still have dependency analysis warnings
 unmigratedProjects=\
-  build-cache-base,build-cache-packaging,hashing,snapshots,\
```
Comment (lptr) [inline gradle.properties:29]: 💭 I took the opportunity to re-enable these checks as part of the PR, too. It doesn't have a big effect, and this way we can be sure that we only have the dependencies we need in the build-cache library projects.

**Review by asodja — APPROVED** [review]
LGTM

**Pull request merged**